build.rs

```diff
@@ -202,6 +202,24 @@ fn main() {
         ccfg.flag_if_supported(flag);
     }
 
+    const PLATFORM_CAN_PROFILE_WITHOUT_FRAME_POINTER: bool = cfg!(any(
+        all(
+            feature = "profile",
+            target_os = "linux",
+            target_arch = "x86_64"
+        ), // Add more platforms here.
+    ));
+
+    fn profile_config(cfg: &mut cc::Build) {
+        if PLATFORM_CAN_PROFILE_WITHOUT_FRAME_POINTER {
+            cfg.debug(true).force_frame_pointer(false);
+        } else if cfg!(feature = "profile") {
+            cfg.debug(true).force_frame_pointer(true);
+        }
+    }
```

Comment:
Sorry for the churn, it occurred to me it might be cleaner
to flip the logic around like so:

```rust
    const PROFILE_BUILD_PLATFORM_REQUIRES_FRAME_POINTER: bool = cfg!(not(any(
        // Whitelist of platforms which do not require frame pointers.
        all(target_os = "linux", target_arch = "x86_64"),
        // Add more platforms here.
    )));
    const PROFILE_BUILD_ENABLED: bool = cfg!(feature = "profile");

    fn profile_config(cfg: &mut cc::Build) {
        if PROFILE_BUILD_ENABLED {
            cfg.debug(true)
                .force_frame_pointer(PROFILE_BUILD_PLATFORM_REQUIRES_FRAME_POINTER);
        }
    }

```

Reply:
No worries, I like it!

Reply:
Hmm, I can't decide whether that means to push that change or not :confused:

Reply:
Sorry for being unclear — I like the proposed change.